packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx

```diff
@@ -0,0 +1,170 @@
+  const context = useFluentContext();
+
+  const { children, className, design, styles, variables, height, estimatedItemSize } = props;
+
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps([...Tree.handledProps, 'estimatedItemSize', 'itemSize'], props);
+
+  const getA11yProps = useAccessibility(props.accessibility, {
+    debugName: VirtualTree.displayName,
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<TreeStylesProps>(Tree.displayName, {
+    className: treeClassName,
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
+
+  const {
+    visibleItemIds,
+    getItemById,
+    registerItemRef,
+    toggleItemActive,
+    focusItemById,
+    expandSiblings,
+    listRef,
+  } = useVirtualTree(props);
+
+  const contextValue: TreeRenderContextValue = React.useMemo(
+    () => ({
+      getItemById,
+      registerItemRef,
+      toggleItemActive,
+      focusItemById,
+      expandSiblings,
+      toggleItemSelect: _.noop,
+    }),
+    [getItemById, registerItemRef, toggleItemActive, focusItemById, expandSiblings],
+  );
+
+  // always use item id as key instead of index (react-window's default)
+  const getItemKey = React.useCallback((index: number, data: VirtualItemData) => data.visibleItemIds[index], []);
+
+  const getItemSize = (index: number) => {
+    const id = visibleItemIds[index];
+    return getItemById(id).item.itemSize || estimatedItemSize;
+  };
+  React.useLayoutEffect(() => {
+    listRef.current.resetAfterIndex(0);
+  }, [listRef, visibleItemIds]); // when item collapsed/expanded (visibleItemIds change), refresh react-window itemSize cache
+
+  const createTreeItem = React.useCallback<VirtualItemData['createTreeItem']>(
+    (id, style) => {
+      const item = getItemById(id);
+      if (item) {
+        const { expanded, parent, level, index, treeSize } = item;
+        return TreeItem.create(item.item, {
+          defaultProps: () =>
+            getA11yProps('item', {
+              renderItemTitle: props.renderItemTitle,
+            }),
+          overrideProps: {
+            style, // came from react-window
+            expanded,
+            parent,
+            key: id,
+            level,
+            index,
+            treeSize,
+            selectable: false,
+          },
+        });
+      }
+      return null;
+    },
+    [getA11yProps, getItemById, props.renderItemTitle],
+  );
+
+  const element = (
+    <TreeContext.Provider value={contextValue}>
+      {getA11yProps.unstable_wrapWithFocusZone(
+        <ElementType
+          {...getA11yProps('root', {
+            className: classes.root,
+            ...rtlTextContainer.getAttributes({ forElements: [children] }),
+            ...unhandledProps,
+          })}
+        >
+          <VariableSizeList
+            width={-1} // width is not used for vertical list
```

Comment:
What happens if you omit the `width` prop ? is there a default set by `react-window` ?

Reply:
When I omit I get typing err because of @types/react-window :( 

There's no default from react-window. 
For list itself, the `width=-1` is passed to React Element is eventually ignored by browser. 
For items within the list, when the list is vertical, 'width=100%' is passed to React Element. 
(related part in react-window is here https://github.com/bvaughn/react-window/blob/d80bef25fe706d0c73fc801674c086f681811190/src/createListComponent.js#L343)